test/unit/common/test_data_descriptor.cpp

```diff
@@ -543,68 +544,102 @@ TYPED_TEST(DataDescriptorTest, CopyCtorFromStridedArray) {
   check_copy_ctor<const TypeParam>(data_const);
 }
 
-template <class TypeParam, class Buffer>
-void check_temporary(Buffer& data) {
+template <class Data>
+void DLAF_CHECK_CREATE_TEMPORARY_BUFFER(const Data& data) {
```

Comment:
I used UPPERCASE name because I thought it would me more clear that they are "on the same level" as `EXPECT_EQ` and other gtest macros. That's the only reason, I can revert it easily if you don't like it.

Reply:
The difference that IMO has to be highlighted, is that the macros can be "line aware" to write the output when an error occour, while functions cannot.
Looking forward to C++20 and `std::source_location` to avoid all this macro non-sense...